File: shared-types/src/api/PlayerCharacter.ts
export interface PlayerCharacter {
  playerCharacterId: number;
  characterName: string;
  alertFeat: boolean;
  luckStone: boolean;
  dexterityMod: number;
  isInPlay: boolean;
  isDeleted: boolean;
}

export interface CreatePlayerCharacterRequest {
  characterName: string;
  alertFeat: boolean;
  luckStone: boolean;
  dexterityMod: number;
}

export interface CreatePlayerCharacterResponse {
  playerCharacterId: number;
}

export interface ReadPlayerCharacterRequest {
  playerCharacterId: number;
}

export interface ReadPlayerCharacterResponse extends PlayerCharacter {}

export interface ReadPlayerCharactersRequest {}

export type ReadPlayerCharactersResponse = PlayerCharacter[];

export interface UpdatePlayerCharacterRequest {
  playerCharacterId: number;
  characterName: string;
  alertFeat: boolean;
  luckStone: boolean;
  dexterityMod: number;
  isInPlay: boolean;
}

export interface DeletePlayerCharacterRequest {
  playerCharacterId: number;
}

File: YetAnotherInitiativeTrackerAPI/Data/Models/PlayerCharacter.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;

public class PlayerCharacter
{
    [BindNever]
    public int PlayerCharacterId { get; set; }
    public int UserId { get; set; }
    public string? CharacterName { get; set; }
    public int DexterityMod { get; set; }
    public bool? LuckStone { get; set; }
    public bool? AlertFeat { get; set; }
    //public bool IsInPlay { get; set; }
    public bool? IsDeleted { get; set; }
}